engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/SafeHtmlValidatorTest.java

```diff
@@ -164,6 +164,20 @@ public void testAdditionalProtocols() {
 		assertNumberOfViolations( validator.validate( new Bar( "<div src='data:image/png;base64,100101' />" ) ), 1 );
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HV-817")
+	public void testPreserveRelativeLinks() throws Exception {
+		descriptor.setValue( "whitelistType", WhiteListType.RELAXED );
+		descriptor.setValue( "baseUri", "http://127.0.0.1" );
+
+		assertTrue( getSafeHtmlValidator().isValid( "<img src='/some/relative/url/image.png' />", null ) );
```

Comment:
It would be nice to also have a test with a real relative URI: `../../some/relative/url/image.png` (one working and one not working if the URI tries to go back too far).